src/batch/src/executor/row_seq_scan.rs

```diff
@@ -259,47 +289,49 @@ impl<S: StateStore> Executor for RowSeqScanExecutor<S> {
 impl<S: StateStore> RowSeqScanExecutor<S> {
     #[try_stream(boxed, ok = DataChunk, error = RwError)]
     async fn do_execute(self: Box<Self>) {
-        match self.scan_type {
-            ScanType::TableScan(iter) => {
-                pin_mut!(iter);
-                loop {
-                    let timer = self.stats.row_seq_scan_next_duration.start_timer();
-
-                    let chunk = iter
-                        .collect_data_chunk(&self.schema, Some(self.chunk_size))
-                        .await
-                        .map_err(RwError::from)?;
-                    timer.observe_duration();
-
-                    if let Some(chunk) = chunk {
-                        yield chunk
-                    } else {
-                        break;
+        for scan_type in self.scan_types {
```

Comment:
Maybe we should create one stream for each `scan_type` and merge them?

Reply:
+1. At least extracting each stream could be more friendly to the diff. 🥵

Reply:
Not sure whether it will improve performance to scan multi ranges at once? cc @hzxa21

Reply:
> Not sure whether it will improve performance to scan multi ranges at once? cc @hzxa21 

Do you mean parallelizing scans on multiple ranges? Technically it can improve performance since we parallelize the I/O.